### PR TITLE
Case contact followups

### LINKED ISSUE #1496: All users can flag a case_contact that requires followup
Part of epic #1495

As **any user**,
I want the ability to flag a `case_contact` that requires followup,
so that the `volunteer` assigned to that case (even if that is myself) knows that further action is required.

### Acceptance Criteria
- [ ] a "Follow up" alert button displays on all `case_contacts` _logged within this quarter_
- [ ] this button displays to the left of the "Edit button" on desktop and underneath it on mobile
- [ ] clicking the "follow up" button affixes an alert symbol to the `case_contact` (see .png in attachments)
- [ ] once a followup has been added to a `case_contact`, the text on the "follow up" button changes to "resolve"

### Attachments
alert symbol (red would be better, but it should look roughly like this)
alert-logo-png-3-Transparent-Images

### PR DESCRIPTION
**Pull request by Thrillberg**
### What github issue is this PR for, if any?
Resolves #1496 

### What changed, and why?
I added the concept of a `Followup` (and its controller with one action, a `create` action). A Followup has a `status` field that can currently be set to either `"requested"` or `"resolved"`. It's an enum, so should be easy enough to add to, if we want. We're also tracking who added the Followup since I think that will come in handy later. A Followup can currently be applied to a CaseContact.

A button as described in the issue was also added. Let me know if you want me to tweak it style-wise!

### How will this affect user permissions?
No effect on permissions.
- Volunteer permissions:
- Supervisor permissions:
- Admin permissions:

### How is this tested? (please write tests!) 💖💪
I wrote tests which I think cover everything. I've also been QAing it as I go (but would welcome more eyes on it!).

### Screenshots please :)
Screen Shot 2021-01-06 at 8 49 04 PM
Screen Shot 2021-01-06 at 8 50 52 PM

### Feelings gif (optional)

accomplished

### CODE REVIEW

**Comment by littleforest** [comment]
I think a `followup_status` enum would be good. Enums are fields that are stored as integers in the database since it takes up less space and has fewer data integrity issues than strings, but you treat them like they are strings or symbols in your code. Plus you get some free methods that make it easy to set the status value. The [documentation](https://api.rubyonrails.org/v5.2.4.4/classes/ActiveRecord/Enum.html) is pretty straightforward.

**Comment by scottolsen** [comment]
Reading through #1495, I'm wondering if a `FollowUp` table would work. If I'm reading it correctly we need not only the follow up status but also who created it. This would also make it easier if creating multiple follow ups for the same case contact becomes a thing. Thoughts?

| follow_ups      |         |
|-----------------|---------|
| id              | bigint  |
| case_contact_id | bigint  |
| status          | integer |
| created_by_id   | bigint  |

I also should have some time tonight after 4 or 5pm central time if you want to pair.

**Comment by Thrillberg** [comment]
Thanks for your help @littleforest and @scottolsen! I've given it another shot here and I think it's a bit closer. @scottolsen I may indeed want to pair tonight but I'm not certain of it yet. I'll post in the Slack channel tonight if I'm up for pairing a bit.

**Comment by compwron** [comment]
don't worry about the web lint build right now, it's having an issue

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "bootstrap-select": "^1.13.18",
     "datatables.net-dt": "^1.10.20",
     "jquery": "^3.5.0",
+    "node-sass": "^4.14.1",
```
Comment (compwron) [inline package.json:22]: We have sass now??

**Comment by compwron** [comment]
I like where this is going! :)

File: package.json (hunk 1 above)
Comment (Thrillberg) [reply package.json:22]: Not intentionally! Not sure how that got there but I will remove it.

**Comment by Thrillberg** [comment]
Methinks this puppy is ready for review!

**Review by compwron — APPROVED** [review]
Very cool!!!

**Pull request merged**